packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -318,4 +375,51 @@ export class DataProxyEngine extends Engine {
       clientVersion: this.clientVersion,
     })
   }
+
+  private async withRetry<T>(args: {
+    callback: (api: { logHttpCall: (url: string) => void }) => Promise<T>
+    actionGerund: string
+  }): Promise<T> {
+    for (let attempt = 0; ; attempt++) {
```

Comment:
Will the for loop with await be completely blocking for all other operations in the runtime?

Reply:
No, it won't, it's basically like a coroutine